ccmlib/scylla_node.py

```diff
@@ -1359,9 +1360,19 @@ def watch_rest_for_alive(self, nodes, timeout=120):
             response = requests.get(url=url_joining)
             if response.text:
                 live = live - set(response.json())
+            # Verify that node knows not only about the existance of the
+            # other node, but also its tokens:
             if tofind.issubset(live):
                 # This node thinks that all given nodes are alive and not
-                # "joining", we're done.
+                # "joining", we're almost done, but still need to verify
+                # that the node knows the others' tokens.
+                check = tofind
+                tofind = set()
+                for n in check:
+                    response = requests.get(url=url_tokens+n)
+                    if response.text == '[]':
+                        tofind.add(n)
```

Comment:
I wonder how this would work with topology-related tests that currently track the log to inject some event during e.g. bootstrap or replace.
Although they can pass `wait_other_notice=False` and implement their own state tracking but they might break with this change.

Reply:
I don't know what kind of tests you mean, but we can see if any tests starts failing with this patch.
I think that if a certain test wants to capture some exact moment during the middle of the boot process, it can't ask ccm to wait until the boot is ove.